hazelcast-jca/src/test/java/com/hazelcast/jca/AbstractDeploymentHzClientTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+* Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package com.hazelcast.jca;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.ResourceAdapterArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+
+import com.hazelcast.core.Hazelcast;
+
+/**
+* base class for jca tests that rely on client connections to hazelcast
+*
+* @author asimarslan
+* @author juanavelez
+*/
+public class AbstractDeploymentHzClientTest {
+    // Start a Hazelcast Instance that will be used by the RAR to connect to in client mode 
+    static {
+        Hazelcast.newHazelcastInstance();
```

Comment:
Both client and server is an instance of HazelcastInstance. Therefore, you shouldn't put any check while cleaning up resource. `stop()` should stay as it was.

```
/* (non-Javadoc)
     * @see javax.resource.spi.ResourceAdapter#stop()
     */
    public void stop() {
        HazelcastInstance instance = hazelcastInstance;
        if (instance != null) {
            instance.getLifecycleService().shutdown();
        }
    }
```

In your test case above, you are launching a new Hazelcast Member outside JEE container so you should start/stop Hazelcast Member using `@BeforeClass/@AfterClass` annotations. 

Resource adapter is only going to shutdown Hazelcast Client if JCA is used in client mode.